### DISCUSSION TITLE
Diskless role vs disklessset tool

### DISCUSSION THREAD

**Discussion by ginomcevoy** [bluebanquise/bluebanquise#589] [Q&A] [answered]
Hello,
I would like to understand the difference between the diskless role at roles/core/diskless (bluebanquise repository) and the disklessset tool (tools repository).

From what I see in the source code, both contain Python code, and they are not in sync. The diskless role for example creates /etc/disklessset/diskless_parameters.yml file, and it looks like the disklessset tool uses that file. But then (if I understand correctly) the diskless role overwrites files provided by the disklessset tool.

Do we need both the role and the tool? How best to use them?
Thanks in advance!

**Comment by oxedions** [comment, marked as answer]
Dear @ginomcevoy 

That is a good question, and answer is: I was lacking time to finalize the move.

Tools are stored in bluebanquise/tools and these are the good ones. They will be provided as rpms.
The disklessset code in the diskless role in bluebanquise/bluebanquise will be removed and the role will simply install the RPM.
Idea is to be able to quickly patch/fix the tools while not having to release a new BlueBanquise version.

Does that make sense to you ?

**Reply by ginomcevoy** [reply]
Hello @oxedions ,
I see that the disklessset tool is making reference to `/etc/disklessset/diskless_parameters.yml` in `disklessset/diskless/image_manager.py`. But this file is only provided by the diskless role/

I want to create a PR to add `/etc/disklessset/diskless_parameters.yml` to `disklesset.spec`. Since it depends on the python version, I was plannning on creating the file inside the SPEC, using cat + %{python3_sitelib} + EOF. What do you think?

**Reply by oxedions** [reply]
Hello @ginomcevoy 

Yes, I think that makes sense. If we need to have the role to update settings in this file later (like images path), we can rely on inline module.
